### ISSUE TITLE
Passing non-null string to nullable ref argument should not warn

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#26739]
This warning is surprising.

```C#
public class C 
{
    public void M() 
    {
        string? s = "";
        M(ref s);  // warning CS8600: Converting null literal or possible null value to non-nullable type.
    }
    void M(ref string? s) => throw null;
}
```

Note: this is also a scenario where we think a special attribute could make sense (this method never sets the `ref` parameter to `null`).

Tagging @cston

**Comment by jcouv** [comment]
Relates to https://github.com/dotnet/roslyn/issues/29958

**Comment by jaredpar** [comment]
Related to #25868
